app/controllers/v1/sessions_controller.rb

```diff
@@ -113,12 +111,10 @@ def saml_settings(options = {})
       SAML::SSOeSettingsService.saml_settings(options)
     end
 
-    def auth_error_code(code)
-      if code == '005' && validate_session
-        UserSessionForm::ERRORS[:saml_replay_valid_session][:code]
-      else
-        code
-      end
+    def saml_error(form)
```

Comment:
what do you think about calling this something like `raise_saml_error`?  It's not obvious below when this is called that it's going to interrupt control flow